src/game/logic/system/gamelogic.cpp

```diff
@@ -62,40 +62,117 @@ bool GameLogic::Is_Intro_Movie_Playing()
     return m_startNewGame && g_theDisplay->Is_Movie_Playing();
 }
 
-int GameLogic::Rebalance_Parent_Sleepy_Update(int update)
+int GameLogic::Rebalance_Parent_Sleepy_Update(int index)
 {
-    captainslog_dbgassert(update >= 0 && (unsigned)update < m_sleepingUpdateModules.size(), "bad sleepy idx");
+    captainslog_dbgassert(index >= 0 && (unsigned)index < m_sleepingUpdateModules.size(), "bad sleepy idx");
 
-    for (int i = ((update + 1) >> 1) - 1; i >= 0; i = ((i + 1) >> 1) - 1) {
-        if (!Compare_Update_Modules(m_sleepingUpdateModules[i], m_sleepingUpdateModules[update])) {
+    for (int i = ((index + 1) >> 1) - 1; i >= 0; i = ((i + 1) >> 1) - 1) {
+
+        UpdateModule *a = m_sleepingUpdateModules[i];

```

Comment:
not sure if this will be functionally equal, a is fetched by index in original b is by i

Reply:
I checked. Implementation looks correct. The ordering is changed, but the indexing and pointers used is the exact same.